modules/compute-api/src/main/java/org/apache/ignite/compute/ComputeJob.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compute;
+
+/**
+ * A Compute job that may be executed on an Ignite node (or a few nodes, or on the whole cluster).
+ *
+ * @param <R> job result type
+ */
+public interface ComputeJob<R> {
```

Comment:
Do we have serializability requirement right now? Just asking

Reply:
It's an interesting question. Here https://issues.apache.org/jira/browse/IGNITE-16613 the `Serializable` is omitted. The current `IgniteCompute` API (namely, the fact that it accepts job class, not job instance) makes me think that we do not need to pass job instances from node to node, instead it seems that each node can instantiate its own instance of a job. In such usage scenario, we don't need a job to be serializable.

Reply:
I think we could leave it as is for now, but ask the guys (Andrey, probably) about the serializability issue.